frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,194 @@
+    };
+
+    const newValue = 'testIdir';
+
+    beforeEach(async () => {
+        wrapper = mount(UserNameInput, {
+            props,
+        });
+
+        usernameInputText = wrapper.find('#userIdInput');
+        usernameInputTextEl = usernameInputText.element as HTMLInputElement;
+        verifyButton = wrapper.find("[data-target-btn='verifyIdir']");
+        verifyButtonEl = verifyButton.element as HTMLButtonElement;
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        wrapper.unmount();
+    });
+
+    it('Should change usernameInput value', async () => {
+        expect(usernameInputTextEl.value).toBe('');
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
+    });
+
+    it('Should receive the correct prop', async () => {
+        //default props
+        expect(wrapper.props()).toEqual(props);
+
+        await wrapper.setProps(newProps);
+        expect(wrapper.props()).toEqual(newProps);
+        expect(wrapper.props()).not.toEqual(props);
+    });
+
+    it('Should call and emit correct value', async () => {
+        await usernameInputText.setValue(newValue);
+        const emitChange = wrapper.emitted('change');
+        expect(emitChange).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult');
+        await wrapper.setProps(newProps);
+        await verifyButton.trigger('click');
+
+        expect(emitSetVerifyResult).toBeTruthy();
+        expect(emitSetVerifyResult![1][0]).toEqual(true);
+    });
+
+    it('Should enable verify btn when username is inputted', async () => {
+        // button starts as disabled
+        expect(verifyButtonEl.disabled).toBe(true);
+        expect(verifyButton.classes('p-disabled')).toBe(true);
+
+        await wrapper.setProps({ userId: newProps.userId });
+
+        expect(verifyButtonEl.disabled).toBe(false);
+        expect(verifyButton.classes('p-disabled')).toBe(false);
+    });
+
+    it('Should show loading on the verify btn while we call the api', async () => {
+        vi.spyOn(
+            AppActlApiService.idirBceidProxyApi,
+            'idirSearch'
+        ).mockImplementation(async () => {
+            setLoadingState(true);
+            return userInputMock(true);
+        });
+
+        setLoadingState(false);
+        expect(verifyButtonEl.textContent).not.toContain('Loading');
+
+        await wrapper.setProps({ userId: newProps.userId });
+        await verifyButton.trigger('click');
+        expect(isLoading()).toBe(true);
+        expect(verifyButtonEl.textContent).toContain('Loading');
+
+        // cleanup state variable
+        setLoadingState(false);
+        expect(isLoading()).toBe(false);
+    });
+
+    it('Should show not found on card when user is not verified', async () => {
```

Comment:
just the title, maybe when user is not found?

Reply:
I'll change to "user is not found"